Xamarin.Forms.Core/ImageElement.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using Xamarin.Forms.Internals;
+
+namespace Xamarin.Forms
+{
+	static class ImageElement
+	{
+
+		public static readonly BindableProperty FileImageProperty = BindableProperty.Create("Image", typeof(FileImageSource), typeof(IImageElement), default(FileImageSource),
+	propertyChanging: OnImageSourceChanging, propertyChanged: OnImageSourceChanged);
+
+		public static readonly BindableProperty SourceProperty = BindableProperty.Create("Source", typeof(ImageSource), typeof(IImageElement), default(ImageSource),
+	propertyChanging: OnImageSourceChanging, propertyChanged: OnImageSourceChanged);
+
+		public static readonly BindableProperty AspectProperty = BindableProperty.Create("Aspect", typeof(Aspect), typeof(IImageElement), Aspect.AspectFit);
+
+		public static readonly BindableProperty IsOpaqueProperty = BindableProperty.Create("IsOpaque", typeof(bool), typeof(IImageElement), false);
+
+
+		static void OnImageSourceChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			ImageSource newSource = (ImageSource)newValue;
+			IImageElement image = (IImageElement)bindable;
+			if (newSource != null && image != null)
+			{
+				newSource.SourceChanged += image.OnImageSourcesSourceChanged;
+			}
+			ImageSourceChanged(bindable, newSource);
+		}
+
+		static void OnImageSourceChanging(BindableObject bindable, object oldValue, object newValue)
+		{
+			ImageSource oldSource = (ImageSource)oldValue;
+			IImageElement image = (IImageElement)bindable;
+
+			if (oldSource != null && image != null)
+			{
+				oldSource.SourceChanged -= image.OnImageSourcesSourceChanged;
+			}
+			ImageSourceChanging(oldSource);
+		}
+
+
+
+		public static SizeRequest Measure(
```

Comment:
I'm uneasy with Measure in the static Element, but that's probably the best place to have it right now